silkworm/silkrpc/core/account_dumper.cpp

```diff
@@ -49,6 +49,10 @@ Task<DumpAccounts> AccountDumper::dump_accounts(
     const auto chain_storage = transaction_.create_storage(tx_database, backend);
 
     const auto block_with_hash = co_await core::read_block_by_number_or_hash(cache, *chain_storage, tx_database, bnoh);
+    if (!block_with_hash) {
+        SILK_ERROR << "dump_accounts:  core::read_block_by_number_or_hash  block not found\n";
+        co_return dump_accounts;
+    }
```

Comment:
I think we need an explicit error handling strategy here (i.e. exception or more specific return value) rather than just returning a valid return value when we detect an error.

Moreover, I don't think we really need to mark this as an error in our logs because it depends on external input, which can be whatever and so this is not really an internal error. I would demote this log trace to `SILK_TRACE`.

Reply:
ok